### PR TITLE
Refactor/reorganize

### PR DESCRIPTION
**Pull request by Niko-sk2x**
Work in progress, submitting to get some discussion on the naming.

I would suggest reviewing individual commits.

### CODE REVIEW

File: src/main/java/io/github/opencubicchunks/cubicchunks/chunk/BigCubeAccess.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import net.minecraft.world.level.levelgen.structure.StructureStart;
 import org.apache.logging.log4j.LogManager;
 
-public interface IBigCube extends BlockGetter, ChunkAccess, FeatureAccess {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/chunk/BigCubeAccess.java]: to `IBigCube` -> `CubeAccess` to match change from `BigCube` -> `LevelCube`

File: src/main/java/io/github/opencubicchunks/cubicchunks/chunk/CubeMap.java [hunk 2]
```diff
@@ -1,24 +1,125 @@
...
+import io.github.opencubicchunks.cubicchunks.chunk.util.CubePos;
+import io.github.opencubicchunks.cubicchunks.utils.Coords;
+import it.unimi.dsi.fastutil.longs.LongSet;
+import net.minecraft.core.SectionPos;
+import net.minecraft.server.level.ChunkHolder;
+import net.minecraft.server.level.ServerChunkCache;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.level.chunk.ChunkStatus;
+
+public interface CubeMap {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/chunk/CubeMap.java]: I sort of like the `I` prefix, though in this case as we're matching vanilla maybe it's fine?

File: src/main/java/io/github/opencubicchunks/cubicchunks/chunk/graph/CubeTicketType.java [hunk 3]
```diff
@@ -6,7 +6,7 @@
 import io.github.opencubicchunks.cubicchunks.mixin.access.common.TicketTypeAccess;
 import net.minecraft.server.level.TicketType;
 
-public class CCTicketType {
+public class CubeTicketType {
     public static final TicketType<CubePos> CCPLAYER = create("player", Comparator.comparingLong(CubePos::asLong));
     public static final TicketType<CubePos> CCFORCED = create("forced", Comparator.comparingLong(CubePos::asLong));
     public static final TicketType<CubePos> CCLIGHT = create("light", Comparator.comparingLong(CubePos::asLong));
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/chunk/graph/CubeTicketType.java]: While we're here, maybe add an `_` to these constants? `CCPLAYER` -> `CC_PLAYER`?

File: src/main/java/io/github/opencubicchunks/cubicchunks/chunk/CubeHolder.java [hunk 4]
```diff
@@ -9,55 +9,55 @@
...
+import io.github.opencubicchunks.cubicchunks.chunk.cube.ImposterProtoCube;
+import io.github.opencubicchunks.cubicchunks.chunk.cube.LevelCube;
 import io.github.opencubicchunks.cubicchunks.chunk.util.CubePos;
 import io.github.opencubicchunks.cubicchunks.mixin.access.common.ChunkHolderAccess;
 import net.minecraft.server.level.ChunkHolder;
 import net.minecraft.server.level.ChunkMap;
 import net.minecraft.world.level.chunk.ChunkStatus;
 
-public interface ICubeHolder {
+public interface CubeHolder {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/chunk/CubeHolder.java]: Removing `I` prefix here

File: src/main/java/io/github/opencubicchunks/cubicchunks/levelgen/chunk/ICubeGenerator.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package io.github.opencubicchunks.cubicchunks.levelgen.chunk;
+
+import io.github.opencubicchunks.cubicchunks.chunk.cube.ProtoCube;
+import io.github.opencubicchunks.cubicchunks.levelgen.CubeWorldGenRegion;
+import net.minecraft.world.level.StructureFeatureManager;
+
+public interface ICubeGenerator {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/levelgen/chunk/ICubeGenerator.java]: Keeping `I` prefix here

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/lighting/LightCubeGetter.java [hunk 6]
```diff
@@ -4,6 +4,6 @@
 
 import net.minecraft.world.level.BlockGetter;
 
-public interface ICubeLightProvider {
+public interface LightCubeGetter {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/world/lighting/LightCubeGetter.java]: `I` prefix inconsistencies

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/lighting/CubicSkyLightEngine.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package io.github.opencubicchunks.cubicchunks.world.lighting;
+
+import io.github.opencubicchunks.cubicchunks.chunk.BigCubeAccess;
+
+public interface CubicSkyLightEngine {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/world/lighting/CubicSkyLightEngine.java:5]: `I` prefix inconsistencies

File: src/main/java/io/github/opencubicchunks/cubicchunks/server/CubicLevelAccessor.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+import javax.annotation.Nullable;
+
+import io.github.opencubicchunks.cubicchunks.chunk.BigCubeAccess;
+import io.github.opencubicchunks.cubicchunks.chunk.util.CubePos;
+import io.github.opencubicchunks.cubicchunks.utils.Coords;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.chunk.ChunkStatus;
+
+public interface CubicLevelAccessor extends CubicLevelHeightAccessor { //TODO: maybe rename this class? I think this name is reserved by the API
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/server/CubicLevelAccessor.java]: `I` prefix inconsistencies

File: src/main/java/io/github/opencubicchunks/cubicchunks/server/ServerCubeCache.java [hunk 9]
```diff
@@ -4,15 +4,15 @@
...
+import io.github.opencubicchunks.cubicchunks.chunk.CubeSource;
+import io.github.opencubicchunks.cubicchunks.chunk.cube.LevelCube;
 import io.github.opencubicchunks.cubicchunks.chunk.util.CubePos;
 import net.minecraft.server.level.ChunkHolder;
 import net.minecraft.server.level.TicketType;
 import net.minecraft.world.level.chunk.ChunkAccess;
 import net.minecraft.world.level.chunk.ChunkStatus;
 
-public interface IServerChunkProvider extends ICubeProvider {
+public interface ServerCubeCache extends CubeSource {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/server/ServerCubeCache.java]: `I` prefix inconsistencies

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/INaturalSpawnerInternal.java [hunk 10]
```diff
@@ -13,9 +13,9 @@
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/world/INaturalSpawnerInternal.java:13]: `I` prefix inconsistencies

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/lighting/SkyLightColumnChecker.java [hunk 11]
```diff
@@ -2,7 +2,7 @@
 
 import io.github.opencubicchunks.cubicchunks.chunk.CubeMapGetter;
 
-public interface ISkyLightColumnChecker {
+public interface SkyLightColumnChecker {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/world/lighting/SkyLightColumnChecker.java:5]: `I` prefix inconsistencies

**Review by NotStirred — COMMENTED** [review]
Note that I'm bringing up inconsistencies, and I don't think them being different is inherently worse. Just want to discuss all of them

inconsistent naming? `Cube` vs `Cubic` in classes: `CubicThreadedLevelLightEngine` `CubicServerLevel` `CubicLayerLightSectionStorage` `CubicLevelLightEngine` `CubicLayerLightEngine` `CubicClientLevel` `CubicLevelAccessor` `CubicSectionStorage` `CubeTracker` `CubeDistanceManager` `CubicPlayerTicketTracker` `FixedPlayerDistanceCubeTracker`

with the only one I sort of think should be changed is `CubeDistanceManager` -> `CubicDistanceManager`

File: src/main/java/io/github/opencubicchunks/cubicchunks/chunk/CubeMap.java (hunk 2 above)
Comment (Niko-sk2x) [reply src/main/java/io/github/opencubicchunks/cubicchunks/chunk/CubeMap.java]: I would like to move away from the I prefix as it is non standard in java and its not used in mc code on forge and fabric anymore

File: src/main/java/io/github/opencubicchunks/cubicchunks/chunk/graph/CubeTicketType.java (hunk 3 above)
Comment (Niko-sk2x) [reply src/main/java/io/github/opencubicchunks/cubicchunks/chunk/graph/CubeTicketType.java]: Fields will be for later. I would just remove the CC prefix from them.

**Comment by Niko-sk2x** [comment]
About inconsistent naming - cubic vs cube - Cube is used instead of Chunk when the original has Chunk in it. Otherwise if there is nothing else to distinguish it, the target would be Cubic. I think I had only one exception where I named something CubicChunk* because just Cube* didn't seem right.

**Comment by Niko-sk2x** [comment]
Resubmitted in new repo https://github.com/OpenCubicChunks/CubicChunks2/pull/26

**Pull request closed without merging**